impress-org/givewp
# Refactor: remove ClassicEditor from window.givewp.form.components and use the new @givewp/form-builder-library npm package

Resolves [GIVE-140](https://stellarwp.atlassian.net/browse/GIVE-140)

## Description



The `ClassicEditor` was [previously](https://github.com/impress-org/givewp/pull/7054) mounted in the window global object to prevent code duplication in the addons. 

But it also introduces a few problems like forcing bumping the minimum Give core version in the addons, and other problems related to possible changes of the components in the core that can break functionalities in the addons. 

~~In the future, a JS package will be released to share the components between the addon without major issues~~

~~But for now, was decided to remove this component from the global window object and fork it in each addon that needs to use it.~~ 

So, this PR removes this component from the global window object and also replaces some components with the ones shipped with the `@givewp/form-builder-library` npm package.

This PR also changes the `OptionsPanel` component used in the donations levels and name title prefixes to the version shipped with the new package.

## Affects



- The `window.givewp.form.components` property.
- Email message content in the email settings of V3 forms.
- The Agreement Text content in the Terms & Consent block of V3 forms.
- The Donation Instructions in the Offline Donations Gateway.
- Donation levels in the Donation Amount and Levels block.
- Name title prefixes in the Donor Name block.

## Testing Instructions



1. Create or edit a v3 form;
2. Go to the email settings and play around with the email message content of the email notifications.
3. Add a Terms & Consent block, go to the Agreement Text content, and play around with the editor options.
4. Go to the Offline Donations gateway settings and play around with the donation instructions content of the gateway.
5. Go to the donation levels in the Donation Amount and Levels block and play around with the options/levels.
6. Go to the name title prefixes in the Donor Name block and play around with the options/prefixes.

To test the addons that were using the ClassicEditor component mounted in the window global object, check out these two PRs:

https://github.com/impress-org/give-tributes/pull/324

https://github.com/impress-org/give-form-field-manager/pull/416

## Pre-review Checklist






[GIVE-140]: https://stellarwp.atlassian.net/browse/GIVE-140?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
